cmd/state-tray/update.go

```diff
@@ -0,0 +1,65 @@
+package main
+
+import (
+	"time"
+
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/pkg/platform/model"
+	"github.com/getlantern/systray"
+	"github.com/gobuffalo/packr"
+)
+
+const (
+	iconUpdateFile      = "icon-update.ico"
+	updateCheckInterval = time.Hour * 24
+)
+
+func superviseUpdate(mdl *model.SvcModel, notice *updateNotice) func() {
+	var done chan struct{}
+
+	go func() {
+		for {
+			notice.show(needsUpdate(mdl))
+
+			select {
+			case <-time.After(updateCheckInterval):
+			case <-done:
+				return
+			}
+		}
+	}()
+
+	return func() { close(done) }
+}
+
+func needsUpdate(mdl *model.SvcModel) bool {
+	for i := 1; i <= 3; i++ {
```

Comment:
This isn't necessary, our graphql client has retrying built-in. YOU implemented that.. ;)